src/shader_compiler.h

```diff
@@ -34,9 +34,13 @@ class ShaderCompiler {
   ~ShaderCompiler();
 
   /// Returns a result code and a compilation of the given shader.
-  /// If the |shader| has a corresponding entry in the |shader_map|, then the
-  /// compilation result is copied from that entry. Otherwise a compiler is
-  /// invoked to produce the compilation result.
+  /// If the shader in |shader_info| has a corresponding entry in the
+  /// |shader_map|, then the compilation result is copied from that entry.
+  /// Otherwise a compiler is invoked to produce the compilation result.
+  ///
+  /// If |shader_info| specifies shader optimizations to run and there is no
```

Comment:
This is unclear what those optimization strings should look like. ...

Please add a comment to src/pipeline.h to the SetShaderOptimizations method to say what that strings should look like. I'd reference the command line options of spirv-opt, or   SPIRV-Tools include/spirv-tools/optimizer.hpp method Optimizer::RegisterPassesFromFlags